src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt

```diff
@@ -0,0 +1,188 @@
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.UnknownPrisonerDetails
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.CourtRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.NotificationRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.ReferenceCodeRepository
+
+@Service
+class RequestBookingService(
+  private val emailService: EmailService,
+  private val contactsService: ContactsService,
+  private val appointmentsService: AppointmentsService,
+  private val courtRepository: CourtRepository,
+  private val prisonRepository: PrisonRepository,
+  private val referenceCodeRepository: ReferenceCodeRepository,
+  private val notificationRepository: NotificationRepository,
+  private val locationsInsidePrisonClient: LocationsInsidePrisonClient,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun request(request: RequestVideoBookingRequest, username: String) {
+    when (request.bookingType!!) {
+      BookingType.COURT -> sendCourtRequestEmails(request, username)
+      BookingType.PROBATION -> log.info("TODO - send probation request emails.")
+    }
+  }
+
+  private fun sendCourtRequestEmails(request: RequestVideoBookingRequest, username: String) {
+    appointmentsService.checkCourtAppointments(request.prisoner().appointments, request.prisoner().prisonCode!!)
+
+    val (pre, main, post) = getCourtAppointments(request.prisoner())
+
+    val prison = prisonRepository.findByCode(request.prisoner().prisonCode!!)
+      ?.also { require(it.enabled) { "Prison with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Prison with code ${request.prisoner().prisonCode} not found")
+
+    val court = courtRepository.findByCode(request.courtCode!!)
+      ?.also { require(it.enabled) { "Court with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Court with code ${request.courtCode} not found")
+
+    val hearingType = referenceCodeRepository.findByGroupCodeAndCode("COURT_HEARING_TYPE", request.courtHearingType!!.toString())!!
+    val locations = locationsInsidePrisonClient.getLocationsByKeys(setOfNotNull(pre?.locationKey, main.locationKey, post?.locationKey)).associateBy { it.key }
+    val contacts = contactsService.getContactsForCourtBookingRequest(court, prison, username).allContactsWithAnEmailAddress()
+
+    contacts.mapNotNull { contact ->
+      when (contact.contactType) {
+        ContactType.OWNER -> createCourtOwnerEmail(contact, request, court, prison, hearingType, main, pre, post, locations)
+        ContactType.PRISON -> createCourtPrisonEmail(contact, request, court, prison, contacts, hearingType, main, pre, post, locations)
+        else -> null
```

Comment:
No COURT contact type?  ie. the primary COURT contact?

Reply:
I have the same question ....

Reply:
But that is a court user making the request in the create booking.  My assumption there might still be wrong though.